### PR TITLE
Feat: add mmr for Players and Observers

### PR DESCRIPTION
**Pull request by NumberPigeon**
should solve issue #89

### CODE REVIEW

File: sc2reader/objects.py [hunk 1]
```diff
@@ -285,6 +285,21 @@ def __init__(self, uid, init_data):
         #: This is deprecated because it doesn't actually work.
         self.recorder = None
 
+        #: The user's mmr at the time of the game
```
Comment (cclauss) [inline sc2reader/objects.py:288]: What is `mmr` and why does the reader need to guess?

File: sc2reader/objects.py [hunk 2]
```diff
@@ -285,6 +285,21 @@ def __init__(self, uid, init_data):
...
+        #: 3. The user has a negative rating, often -36400.
+        #: For ease of use, this property will return None in both cases.
+        if (
+            "scaled_rating" in init_data
+            and init_data["scaled_rating"] is not None
+            and init_data["scaled_rating"] > 0
+        ):
+            self.mmr = init_data["scaled_rating"]
+        else:
+            self.mmr = None
```
Comment (cclauss) [inline sc2reader/objects.py]:
```suggestion
        matchmaking_rating = int(init_data.get("scaled_rating") or 0)
        self.mmr = matchmaking_rating if matchmaking_rating > 0 else None
```

**Review by cclauss — CHANGES_REQUESTED** [review]

File: sc2reader/objects.py (hunk 2 above)
Comment (NumberPigeon) [reply sc2reader/objects.py]: Thanks for the review and suggestion!

File: sc2reader/objects.py (hunk 1 above)
Comment (NumberPigeon) [reply sc2reader/objects.py:288]: MMR stands for Matchmaking Rating, indicating the skill level on SC2 ladder. I assume this is a common abbreviation like APM under sc2 context, but now I dont know whether we should explain it in comment or not.

File: sc2reader/objects.py [hunk 3]
```diff
@@ -285,6 +285,15 @@ def __init__(self, uid, init_data):
...
         self.recorder = None
 
+        #: The user's mmr at the time of the game
+        #: Currently, there are three cases observed for a user that does not have a current mmr:
+        #: 1. The user has no 'scaled_rating' key in their init_data,
+        #: 2. The user has a None value for their 'scaled_rating' key, or
+        #: 3. The user has a negative rating, often -36400.
+        #: For ease of use, this property will return None in both cases.
+        mmr = int(init_data.get("scaled_rating") or 0)
+        self.mmr = mmr if mmr > 0 else None
```
Comment (cclauss) [inline sc2reader/objects.py]:
```suggestion
        matchmaking_rating = int(init_data.get("scaled_rating") or 0)
        self.mmr = matchmaking_rating if matchmaking_rating > 0 else None
```

**Review by cclauss — APPROVED** [review]
LGTM

**Review by StoicLoofah — APPROVED** [review]
thanks for the detailed comment!

**Pull request merged**